wire_protocol/src/connection_factory.rs

```diff
@@ -99,8 +99,28 @@ impl ConnectionFactory for TokioConnectionFactory {
                 endpoint
             );
             let mut config = rustls::ClientConfig::new();
-            let mut pem = BufReader::new(File::open(&self.path).expect("open pem file"));
-            config.root_store.add_pem_file(&mut pem).expect("add pem file");
+            let cert_paths =
+                std::fs::read_dir(&self.path).expect("cannot read from the provided cert directory");
+            for entry in cert_paths {
+                let path = entry.expect("get the cert file").path();
+                debug!("reading cert file {}", path.display());
+                let mut pem = BufReader::new(File::open(path.clone()).expect("read cert file"));
```

Comment:
Same thing here. We should avoid reading files everytime we establish a connection. It would be better to do this once per factory.